src/tests/match-requests.spec.ts

```diff
@@ -115,7 +118,43 @@ describe("Match requests", () => {
         key3: "c"
       },
       field2: "d",
-      field3: 1
+      field3: 1,
+      requestCount: 2,
+    });
+  });
+
+  describe('When two requests are the same', () => {
+    // @ts-ignore
+    beforeAll(async () => {
+      await axios.post(`${PROXAY_HOST}/__proxay/tape`, {
+        tape: "tape",
+        mode: "record"
+      });
+
+      await axios.post(`${PROXAY_HOST}${JSON_IDENTITY_PATH}`, {
+        field3: 1,
+      });
+
+      await axios.post(`${PROXAY_HOST}${JSON_IDENTITY_PATH}`, {
+        field3: 1,
+      });
+    });
+
+    it("Picks the request that hasn't been matched over the request has been matched", async () => {
+      await axios.post(`${PROXAY_HOST}/__proxay/tape`, {
+        tape: "tape",
+        mode: "replay"
+      });
+
+      const response1 = await axios.post(`${PROXAY_HOST}${JSON_IDENTITY_PATH}`, {
+        field3: 1
+      });
+
+      const response2 = await axios.post(`${PROXAY_HOST}${JSON_IDENTITY_PATH}`, {
+        field3: 1
+      });
```

Comment:
Could you add a third response to see how it behaves when there is one more request than expected?

Reply:
Yeah. Will do.

Reply:
Added another test below to show the behaviour:
The 3rd response will be the same as the last one. 
I thought about potentially this may need to be an exception.
But that falls outside of this change IMO.

Reply:
I'd prefer that we don't throw an exception. It's entirely possible to have a frontend that sends a request an unspecified number of times. This behaviour actually sounds perfect :)